web_widget_open_tab/static/src/js/widget.js

```diff
@@ -0,0 +1,82 @@
+odoo.define('web_widget_open_tab.FieldOpenTab', function(require) {
+    "use strict";
+
+    var AbstractField = require('web.AbstractField');
+    var field_registry = require('web.field_registry');
+    var ListRenderer = require('web.ListRenderer');
+    var core = require('web.core');
+    var config = require('web.config');
+    var qweb = core.qweb;
+    var _t = core._t;
+
+    var FieldOpenTab = AbstractField.extend({
+        description: "",
+        supportedFieldTypes: ['integer'],
+        events: _.extend({}, AbstractField.prototype.events, {
+            'click': '_onClick',
+        }),
+        isSet: function () {
+            return true;
+        },
+        _getReference: function () {
+            var url = new URL(window.location.href);
+            var search = 'model=' + this.model;
+            var re = new RegExp('^.*' + search + '$')
+            if (
+                url.hash.search(search + '&') === -1 &&
+                url.hash.search(re) === -1
+            ) {
+                return '#view_type=model&model=' + this.model + '&id=' + this.res_id;
+            }
+            return url.hash.replace(/view_type=\w+/i, "view_type=form") + '&id=' + this.res_id;
+        },
+        _renderReadonly: function () {
+            var $content = $(
+                '<a href="'+ this._getReference() + '">'
+            ).addClass('open_tab_widget fa fa-eye');
+            var self = this;
+            $content.tooltip({
+                delay: { show: 1000, hide: 0 },
+                title: function () {
+                    return qweb.render('WidgetButton.tooltip', {
+                        debug: config.debug,
+                        state: self.record,
+                        node: {
+                            attrs: {
+                                'help': _t('Click in order to open on new tab'),
+                                'type': _t('Widget')
+                            }
+                        },
+                    });
+                },
+            });
+            this.$el.append($content)
+        },
+        _onClick: function (ev) {
+            console.log(this);
+            console.log(ev);
+            ev.preventDefault();
+            ev.stopPropagation();
+            var element = $(ev.currentTarget).find('a');
```

Comment:
can't you check the tagName?

Reply:
I prefer to check in order to ensure that we have not selected any other element it is just for security. I have changed it by closest